Lib/zipfile/_path.py

```diff
@@ -254,12 +254,12 @@ def open(self, mode='r', *args, pwd=None, **kwargs):
             raise FileNotFoundError(self)
         stream = self.root.open(self.at, zip_mode, pwd=pwd)
         if 'b' in mode:
-            if args or kwargs:
+            if encoding is not None or args or kwargs:
```

Comment:
I wish the change didn't add extra (cyclomatic) complexity to the implementation. For an interface that's meant to be pass-through, it's unfortunate that the 'encoding' concern is so entangled with the wrapper behavior. Now these methods need to repeat the default value (`None`) and consider quite a few more cases. At this point, I'm wondering of the handling of "transform encoding parameter" should be extracted to its own wrapper to capture its nuances.

Reply:
A test failure pointed me at the need for this. Unfortunately adding this check is the only way to maintain the "any other parameters cause a ValueError" behavior of the existing API.

We don't really have a choice about adding 'encoding' to the wrappers direct API given we need to call `io.text_encoding` on it no matter which way it was passed in to maintain the behavior that added io.text_encoding in the first place.

Reply:
this is no longer needed with the latest revision.